clang/test/CodeGen/inline-asm-systemz-flag-output.c

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -O2 -triple s390x-linux -emit-llvm -o - %s | FileCheck %s
+
+int foo_012(int x) {
+// CHECK-LABEL: @foo_012
+// CHECK: = tail call { i32, i32 } asm "ahi $0,42\0A", "=d,={@cc},0"(i32 %x)
+  int cc;
+  asm ("ahi %[x],42\n" : [x] "+d"(x), "=@cc" (cc));
+  return cc == 0 || cc == 1 || cc == 2 ? 42 : 0;
```

Comment:
This doesn't actually verify any of the cc == 0 etc. logic in the generated IL.  And in fact, I don't think there's any point in doing so, but then there's no point in having it in the source either.   Why not just have a plain test case that does a "return cc" ?   There's no need in having multiple versions of the test if you don't even look at any of the differences either.

However, it would be good to validate the full IL output - specifically, that we generate the correct assertion.